mqemitter-redis.js

```diff
@@ -146,19 +148,20 @@ MQEmitterRedis.prototype.on = function on (topic, cb, done) {
   return this
 }
 
-MQEmitterRedis.prototype.emit = function (msg, done) {
+MQEmitterRedis.prototype.emit = function (msg, done, packet = null) {
   done = done || this._onError
 
   if (this.closed) {
     var err = new Error('mqemitter-redis is closed')
     return done(err)
   }
 
-  var packet = {
-    id: hyperid(),
-    msg: msg
+  if (packet == null) {
+    packet = {
+      id: hyperid(),
+      msg: msg
+    }
```

Comment:
why is this change needed?

Reply:
to help test the error, so i can emit mqtt with non object payload

Reply:
oh, looks like i wrong here, please don't merge the test, i'll fix it

Reply:
![image](https://user-images.githubusercontent.com/3325901/96138479-03b16d00-0f28-11eb-91bc-bed13525d841.png)
looks like topic and message from pmessageBuffer cannot decoded using msgpack.decode, but Buffer.from can. don't know how the encoding can be different. not sure how to fix this :(

Reply:
oh, got it, so it happened because i'm having Queue using Kue, and aedes as broker. both using same redis server, but the redis database is different (one using db:1, while the other db:2) , the question is why even it's different but the message is received by other. and this is not happened if the subscriber not using # as topic.

Reply:
I do not understand the issue or how this could happen.

Reply:
sorry, please close this and don't need to merge the changes, the issue is not in mqemitter-redis but from ioredis. when wildcard used is *, it then read events from other redis db. if redis used not only for mqtt but for something else, in my case it's queue then the message received in your mqttemitter is possible from queue.